### PR TITLE
[json-rpc] validate version params, return client error code instead of internal error code

### PR DESCRIPTION
**Pull request by xli**
1. added test to cover error cases, and enforce return code -32602
2. created parse_version_param to handle version param parsing and validation.

### CODE REVIEW

**Comment by libra-action** [comment]
**This PR may have modified JSON-RPC server code.**

Breaking changes policy:

1. Chaning JSON-RPC method signatures or removing/modifying fields
   in the response in /V1 will not be accepted.
2. Adding fields is ok.
3. Adding a new JSON-RPC method is ok.

Please ensure you have also done the following to update the docs:

1. Update json-rpc/API-CHANGELOG.md.
2. Add/update type/method documentation under json-rpc/docs.

File: json-rpc/src/tests/unit_tests.rs [hunk 1]
```diff
@@ -277,6 +277,22 @@ fn test_json_rpc_protocol_invalid_requests() {
                 "libra_ledger_version": version
             }),
         ),
+        (
+            "get_metadata invalid arguments: version is too large",
```
Comment (thefallentree) [inline json-rpc/src/tests/unit_tests.rs:281]: too high?

Comment (thefallentree) [reply json-rpc/src/tests/unit_tests.rs:281]: out of range?

**Review by thefallentree — APPROVED** [review]

**Pull request closed without merging**